trustfall_core/src/interpreter/hints/candidates.rs

```diff
@@ -28,19 +28,6 @@ pub enum CandidateValue<T> {
     All,
 }
 
-impl<T> CandidateValue<T> {
```

Comment:
We don't want to return back `CandidateValue<&T>` anymore so this isn't necessary anymore

Reply:
I think either leave `.as_ref()` or also remove `.cloned()` in the impl below. They are duals of each other.

Reply:
I can't remove `.cloned()` as we still internally use `CandidateValue<&FieldValue>`, but before we return it, we call `.cloned()` to remove the reference, see: https://github.com/obi1kenobi/trustfall/pull/466/files#diff-84f518e6512bfe60e05ad1f0c0fe9a9f63e4e0220a652aa15d07c1af6a4e24f1R229

Reply:
I did just make it pub(crate) though, as we should be the only consumers of this API after this change.

Reply:
sounds good!